azure/datalake/store/transfer.py

```diff
@@ -361,6 +362,27 @@ def progress(self):
                 chunks=chunks,
                 exception=self._files[key]['exception']))
         return files
+    
+    def _rename_file(self, src, dst, overwrite=False):
+        """ Rename a file from file_name.inprogress to just file_name. Invoked once download completes on a file.
+
+        Internal function used by `download`.
+        """
+        try:
+            # we do a final check to make sure someone didn't create the destination file while download was occuring
+            # if the user did not specify overwrite.
+            if os.path.isfile(dst):
+                if not overwrite:
+                    raise FileExistsError(dst)
+                os.remove(dst)
+            os.rename(src, dst)
+        except Exception as e:
+            exception = repr(e)
+            logger.error('Rename failed for source file: %s; %s', src, exception)
+            raise e
+    
+        logger.debug('Renamed %s to %s', src, dst)
```

Comment:
Use `%r` in case their is unforeseen issues with some wonky whitespace used in the file path.